modules/ai/assets/js/editor/pages/form-layout/hooks/use-screenshots.js

```diff
@@ -34,6 +35,17 @@ const useScreenshots = ( { onData } ) => {
 				return updatedData;
 			} );
 
+			setPromptPrevIds( ( prev ) => {
+				if ( ! prev[ prompt ] ) {
+					prev[ prompt ] = [];
+				}
+
+				return {
+					...prev,
+					[ prompt ]: [ ...prev[ prompt ], screenshot.baseTemplateId ],
+				};
+			} );
```

Comment:
can't you make it a derived state of the screenshots array?

```
const prevIds = screenshots.map( ( s ) => s.baseTemplateId )
```

Or am I missing something?

Reply:
Fixed